cpp/bench/ann/src/cuvs/cuvs_benchmark.cu

```diff
@@ -71,6 +108,30 @@ auto create_algo(const std::string& algo_name,
     parse_build_param<T, uint32_t>(conf, param);
     a = std::make_unique<cuvs::bench::cuvs_cagra<T, uint32_t>>(metric, dim, param);
   }
+#endif
+#ifdef CUVS_ANN_BENCH_USE_CUVS_MG
+
+  if (algo_name == "raft_mg_ivf_flat" || algo_name == "cuvs_mg_ivf_flat") {
+    typename cuvs::bench::cuvs_mg_ivf_flat<T, int64_t>::build_param param;
+    parse_build_param<T, int64_t>(conf, param);
+    add_distribution_mode(&param.mode, conf);
+    a = std::make_unique<cuvs::bench::cuvs_mg_ivf_flat<T, int64_t>>(metric, dim, param);
+  }
+
+  if (algo_name == "raft_mg_ivf_pq" || algo_name == "cuvs_mg_ivf_pq") {
+    typename cuvs::bench::cuvs_mg_ivf_pq<T, int64_t>::build_param param;
+    parse_build_param<T, int64_t>(conf, param);
+    add_distribution_mode(&param.mode, conf);
+    a = std::make_unique<cuvs::bench::cuvs_mg_ivf_pq<T, int64_t>>(metric, dim, param);
+  }
+
+  if (algo_name == "raft_mg_cagra" || algo_name == "cuvs_mg_cagra") {
+    typename cuvs::bench::cuvs_mg_cagra<T, uint32_t>::build_param param;
+    parse_build_param<T, uint32_t>(conf, param);
```

Comment:
Here (and above in the mg cases) can't we just parse the single GPU param struct, and construct the parallel param struct from that?